internal/controller/tortoise_controller.go

```diff
@@ -238,6 +239,13 @@ func (r *TortoiseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		logger.Error(err, "failed to get HPA", "tortoise", req.NamespacedName)
 		return ctrl.Result{}, err
 	}
+	scalingActive := r.HpaService.CheckHpaMetricStatus(ctx, hpa)
+	if scalingActive == false && tortoise.Spec.UpdateMode == autoscalingv1beta3.UpdateModeAuto && tortoise.Status.TortoisePhase == autoscalingv1beta3.TortoisePhaseWorking {
+		//switch to emergency mode only when auto tortoise and already working
+		tortoise.Spec.UpdateMode = autoscalingv1beta3.UpdateModeEmergency
```

Comment:
@sanposhiho somehow this line doesn't change the actual spec that k8sclient.get retrieves so in the test while tortoisephase changes to emergency, spec.updatemode is still auto. If this is the case, i think we need to create a new tortoisephase for auto scale up. What do you think?

Reply:
> somehow this line doesn't change the actual spec

because `UpdateTortoiseStatus` below only updates .status (literally), not .spec.
And, thinking about it, probably we shouldn't directly change the spec. It's breaking the boundary between the user's intent and the controller.

So,

> If this is the case, i think we need to create a new tortoisephase for auto scale up.

yeah. I think, instead of changing the updatemode on spec, we should change something at .status. `TortoisePhase` makes sense to me.

Reply:
Can we just use the existing `TortoisePhaseEmergency`, not a new one?

Reply:
Actually i am not sure what other scenarios will require emergency mode. If missing metrics is the only scenario then it would make sense to disallow manual switch. What do you think?

Reply:
> if we use existing emergency phase, when tortoise is in auto mode it will reconcile and switch it back to working right?

No, it shouldn't. That depends on how you implement. The controller should keep putting `TortoisePhaseEmergency` as long as HPA has a missing metric, even if the mode is auto.

> Actually i am not sure what other scenarios will require emergency mode. If missing metrics is the only scenario then it would make sense to disallow manual switch. What do you think?

There could be many scenarios right now, e.g.,
- Mercari is going to have a Black Friday sale, and we want to pre-autoscale the number of replicas. (could be solved by https://github.com/mercari/tortoise/pull/370)
- A big upstream microservice is down, and the downstream microservice is getting traffic much less than usual now. The dynamic minReplicas feature keeps the replica number to some extent though, we want to make it bigger enough to prepare the big upstream microservice to come back fully.
- We're getting requests more than usual, unexpectedly (DDoS, etc), and the platform needs to scale up all the services.

I agree that we should enhance tortoise so that users don't have to use emergency mode at all _eventually_, but for now, we should have it, as a last resort.